### ISSUE TITLE
Move `relstorage/` into a `src/` directory

### ISSUE THREAD

**Issue by jamadden** [zodb/relstorage#218]
That seems to be the modern best practice. Essentially all `zopefoundation` repos do this.

It does complicate git history, though, and would make supporting multiple versions (before and after the switch) more complicated because patches won't necessarily cleanly apply.

**Comment by jamadden** [comment]
Personally, I prefer code in `src/` because I have a bunch of shell aliases set up that assume that (e.g., shortcuts for running tests `alias ztest = zope.testrunner --test-path=src`). Plus I think it just looks much neater 🤷‍♂ 

I have run into trouble in the past when `.` is on the Python path and the working directory is the checkout, causing it to pick up an unexpected version of the code.

**Comment by jamadden** [comment]
> I have run into trouble in the past when . is on the Python path and the working directory is the checkout, causing it to pick up an unexpected version of the code.

Heh, I just ran into that today with this project. I created a new virtualenv to test a different Python version with and started running relstorage tests. When I looked closely, I noticed output like this:
```
//relstorage/tests/test_zodburi.py:170: UserWarning: relstorage not found, skipping the zodburi test for postgres
```

That's *supposed* to be a warning when we can't get a database driver, but instead the `relstorage` distribution itself was triggering them. Turns out I'd forgotten to install the project, but because `.` was on the path, many of the tests worked.
